spec/std/path_spec.cr

```diff
@@ -0,0 +1,594 @@
+  when String then posix = Path.posix(posix)
+  when Array  then posix = posix.map { |path| Path.posix(path) }
+  end
+  case windows
+  when Nil
+  when Tuple  then windows = Path.windows(*windows)
+  when String then windows = Path.windows(windows)
+  when Array  then windows = windows.map { |path| Path.windows(path) }
+  end
+  assert_paths_raw(path, posix, windows, label, file, line, &block)
+end
+
+private def assert_paths_raw(path, posix, windows = posix, label = nil, file = __FILE__, line = __LINE__, &block : Path -> _)
+  it %(#{label} "#{path}" (posix)), file, line do
+    block.call(Path.posix(path)).should eq(posix)
+  end
+  it %(#{label} "#{path}" (windows)), file, line do
+    block.call(Path.windows(path)).should eq(windows)
+  end
+end
+
+describe Path do
+  describe ".new" do
+    it { Path.new("foo").native?.should be_true }
+    it { Path.new("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.new("foo\0")
+      end
+    end
+
+    it { Path.new.to_s.should eq "" }
+  end
+
+  describe ".posix" do
+    it { Path.posix("foo").posix?.should be_true }
+    it { Path.posix("foo").windows?.should be_false }
+    it { Path.posix("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.posix("foo\0")
+      end
+    end
+
+    it { Path.posix.to_s.should eq "" }
+  end
+
+  describe ".windows" do
+    it { Path.windows("foo").posix?.should be_false }
+    it { Path.windows("foo").windows?.should be_true }
+    it { Path.windows("foo").to_s.should eq "foo" }
+
+    it "fails with null byte" do
+      expect_raises ArgumentError, "String contains null byte" do
+        Path.windows("foo\0")
+      end
+    end
+
+    it { Path.windows.to_s.should eq "" }
+  end
+
+  describe "{}" do
+    Path{"foo"}.should eq Path.new("foo")
+    Path{"foo"}.native?.should be_true
+  end
+
+  describe "literal" do
+    it { Path{"foo", "bar", "baz"}.should eq Path.new("foo", "bar", "baz") }
+    it { Path{"/foo", "bar", "baz"}.should eq Path.new("/foo", "bar", "baz") }
+  end
+
+  describe "#parent" do
+    assert_paths("/Users/foo/bar.cr", "/Users/foo", &.parent)
+    assert_paths("Users/foo/bar.cr", "Users/foo", &.parent)
+    assert_paths("foo/bar/", "foo", &.parent)
+    assert_paths("foo/bar/.", "foo/bar", &.parent)
+    assert_paths("foo/bar/..", "foo/bar", &.parent)
+    assert_paths("foo", ".", &.parent)
+    assert_paths("foo/", ".", &.parent)
+    assert_paths("/", "/", &.parent)
+    assert_paths("////", "/", &.parent)
+    assert_paths("foo//.//", "foo", &.parent)
+    assert_paths("/.", "/", &.parent)
+    assert_paths("/foo", "/", &.parent)
+    assert_paths("", ".", &.parent)
+    assert_paths("./foo", ".", &.parent)
+    assert_paths(".", ".", &.parent)
+    assert_paths("\\Users\\foo\\bar.cr", ".", "\\Users\\foo", &.parent)
```

Comment:
I'm a bit lost why the parent of the windows paths in linux return `.`.

Reply:
`\\Users\\foo\\bar.cr` is a file name on Linux, exactly like `foo`. The parent is `.`